client/src/views/devices/ExportRecoveryDevicePage.vue

```diff
@@ -0,0 +1,204 @@
+                  >
+                    {{ $t('RecoveryDevicePage.actions.download') }}
+                  </ion-button>
+                </div>
+                <div v-else>
+                  <ion-icon
+                    :icon="checkmarkCircle"
+                    class="checked"
+                  />
+                  {{ $t('RecoveryDevicePage.subtitles.fileDownloaded') }}
+                </div>
+              </div>
+              <div class="block">
+                {{ $t('RecoveryDevicePage.titles.recoveryKey') }}
+                <div v-if="!recoveryKeyDownloaded">
+                  <ion-button
+                    @click="downloadKey()"
+                  >
+                    {{ $t('RecoveryDevicePage.actions.download') }}
+                  </ion-button>
+                </div>
+                <div v-else>
+                  <ion-icon
+                    :icon="checkmarkCircle"
+                    class="checked"
+                  />
+                  {{ $t('RecoveryDevicePage.subtitles.fileDownloaded') }}
+                </div>
+              </div>
+              <a
+                ref="downloadLink"
+                v-show="false"
+              />
+              <div v-if="recoveryKeyDownloaded && recoveryFileDownloaded">
+                <ion-button
+                  class="return-btn button-outline"
+                  @click="routerNavigateTo('workspaces')"
+                >
+                  <ion-icon
+                    :icon="home"
+                    class="icon"
+                  />
+                  {{ $t('RecoveryDevicePage.actions.backToWorkspaces') }}
+                </ion-button>
+              </div>
+            </div>
+          </div>
+        </div>
+      </div>
+    </ion-content>
+  </ion-page>
+</template>
+
+<script setup lang="ts">
+import {
+  IonPage,
+  IonContent,
+  IonButton,
+} from '@ionic/vue';
+import MsInformativeText from '@/components/core/ms-text/MsInformativeText.vue';
+import { ref, inject } from 'vue';
+import { getPasswordFromUser } from '@/components/core/ms-modal/MsPasswordInputModal.vue';
+import { useI18n } from 'vue-i18n';
+import { exportRecoveryDevice, RecoveryDeviceErrorTag } from '@/parsec';
+import { NotificationManager, Notification, NotificationKey, NotificationLevel } from '@/services/notificationManager';
+import { routerNavigateTo } from '@/router';
+import { home, checkmarkCircle, fingerPrint } from 'ionicons/icons';
+
+const { t } = useI18n();
+
+enum ExportDevicePageState {
+  Start = 'start',
+  Download = 'download',
+}
+
+const state = ref(ExportDevicePageState.Start);
+let code = '';
+let file = '';
+const downloadLink = ref();
+const recoveryKeyDownloaded = ref(false);
+const recoveryFileDownloaded = ref(false);
+const notificationManager: NotificationManager = inject(NotificationKey)!;
+
+async function exportDevice(): Promise<void> {
+  const password = await getPasswordFromUser({
+    title: t('PasswordInputModal.passwordNeeded'),
```

Comment:
If these texts are generic and don't apply only in this case, might as well put them as default in `MsPasswordInputModal` and not have them as arguments.